libs/utils/utils.js

```diff
@@ -1295,3 +1297,32 @@ export function loadLana(options = {}) {
 }
 
 export const reloadPage = () => window.location.reload();
+
+export const userCanPublishPage = async (detail, isBulk = true) => {
+  if (!detail) return false;
+  const { live, profile, webPath } = detail;
+  let canPublish = isBulk ? live?.permissions?.includes('write') : true;
+  let message = 'Publishing is currently disabled for this page';
+  const config = await getCustomConfig('/.milo/publish-permissions-config.json');
+  const item = config?.urls?.data?.find(({ url }) => (
+    url.endsWith('**') ? webPath.includes(url.slice(0, -2)) : url === webPath
+  ));
+  if (item) {
+    canPublish = false;
+    if (item.message) message = item.message;
+    const group = config[item.group];
+    if (group && profile?.email) {
+      let isDeny;
+      const user = group.data?.find(({ allow, deny }) => {
+        if (deny) {
+          /* c8 ignore next 3 */
+          isDeny = true;
+          return deny === profile.email;
+        }
+        return allow === profile.email;
+      });
+      canPublish = isDeny ? !user : !!user;
+    }
+  }
+  return { canPublish, message };
+};
```

Comment:
Any chance this can be moved to the tools utils or some other file? Ability to publish things should only be relevant for authoring purposes, not for our users.

Reply:
I moved this to the `tools/utils` dir.